### ISSUE TITLE
Chapter16 (16.5.4) Finetuning using the Trainer API

### ISSUE THREAD

**Issue by Shunsuke1015** [rasbt/machine-learning-book#109]
Please forgive the discrepancy in page numbers, as I am reading a book translated into Japanese.

I was in Code (p.543 top row) fine tuning DistliBert using Huggingface/Transformer's Trainer, and I found the
```
Trainer(...
    eval_dataset=test_dataset
)

```
and after the training is finished, the evaluation is done with trainer.evaluate().
At the start of 16.5.1, the Train/Validation/Test dataset was created from the IMDb dataset, and
When training directly with Pytorch (16.5.3), the evaluation of each Epoch is done with valid_dataloader and the last evaluation is done with test_dataloader, so
In the training-evaluation by Trainer, the
```
Trainer(...
    eval_dataset=valid_dataset
)
```
and the evaluation is done in
`trainer.evaluate(eval_dataset=test_dataset)`
would be more correct?

**Comment by rasbt** [comment]
Yes, that's a good point. I am more familiar with the Lightning Trainer (https://github.com/rasbt/deeplearning-models/blob/master/pytorch-lightning_ipynb/transformer/distilbert-finetuning-ii.ipynb) which the original HuggingFace Trainer was based on. I assumed that the eval_dataset was referring to the validation set -- the same is a bit ambiguous.

**Comment by Shunsuke1015** [comment]
Thanks for your comment. As you suggested, in Pytorch lightning I can set train, valid, and test arguments, but in Huggingface Trainer I can only set Train and Eval.  It was very informative.
